plugins/glustershd/rest.go

```diff
@@ -265,3 +267,95 @@ func selfHealHandler(w http.ResponseWriter, r *http.Request) {
 	}
 	restutils.SendHTTPResponse(ctx, w, http.StatusOK, nil)
 }
+
+func splitBrainOperationHandler(w http.ResponseWriter, r *http.Request) {
+	// Collect inputs from URL
+	volname := mux.Vars(r)["volname"]
+	operation := mux.Vars(r)["operation"]
+
+	ctx := r.Context()
+	logger := gdctx.GetReqLogger(ctx)
+
+	var req glustershdapi.SplitBrainReq
+	if err := restutils.UnmarshalRequest(r, &req); err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, gderrors.ErrJSONParsingFailed)
+		return
+	}
+
+	// check if hostname is in form of uuid, if yes, convert uuid to respective ip addr
+	if uuid.Parse(req.HostName) != nil {
+		if peers, err := peer.GetPeer(req.HostName); err == nil {
```

Comment:
this can be 
```
if peers, err := peer.GetPeer(req.HostName); err != nil {
	restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, "couldn't find ip for the peerid")
        return
}
req.HostName = strings.Split(peers.PeerAddresses[0], ":")[0]

```